src/basic/chase.c

```diff
@@ -651,7 +651,7 @@ int chase(const char *path, const char *root, ChaseFlags flags, char **ret_path,
 
                                 assert(!path_is_absolute(p));
 
-                                q = path_join(root, p + (*p == '.'));
+                                q = path_join(root, p + STR_IN_SET(p, ".", "./"));
```

Comment:
If I understand correctly, the previous code was relying on the bool being 1 and thus skipping over the dot, but for the ./ case it means it doesn't skip over the / too, is that intentional?

Reply:
`path_join()` will merge multiple slashes into one